### ISSUE TITLE
Activation with WP-CLI fails to set up view_query_monitor capability

### ISSUE THREAD

**Issue by johnbillion** [johnbillion/query-monitor#142]
[As reported in the support forums](https://wordpress.org/support/topic/wp-cli-activation-doesnt-seem-to-attach-the-admin-menu).

Activating QM with WP-CLI does not result in the `view_query_monitor` capability being assigned to admins.

Either WP-CLI isn't triggering the activation hook, or QM's activation hook isn't compatible with what happens when WP-CLI does fire it.

**Comment by CaliperCreative** [comment]
Hey I did a little testing and it seems like the difference of the WP-CLI activate and "manual" activation is the `db.php` file being symlinked. John, does this help identify where the init sequence is stalling?

**Comment by rask** [comment]
I was wondering why the plugin did not show the toolbar menu after `wp-cli plugin activate query-monitor`. Any news on this one? I'm running on WP 4.4.1 and WP-CLI 0.22.0. QM is the only plugin I have installed (and CLI-activated).

**Comment by phh** [comment]
Guess this is the reason:
```
if ( 'cli' === php_sapi_name() && ! defined( 'QM_TESTS' ) ) {
	# For the time being, let's not load QM when using the CLI because we've no persistent storage and no means of
	# outputting collected data on the CLI. This will hopefully change in a future version of QM.
	return;
}
```

Another check for WP_CLI could be added:
```
if ( 'cli' === php_sapi_name() && ! defined( 'QM_TESTS' ) && ( ! defined( 'WP_CLI' ) || ! WP_CLI ) ) {
	# For the time being, let's not load QM when using the CLI because we've no persistent storage and no means of
	# outputting collected data on the CLI. This will hopefully change in a future version of QM.
	return;
}
```

Or something like only registering the activate/deactivate hooks during CLI. I could open a PR. What do you think?

**Comment by johnbillion** [comment]
I'm pretty sure that's a red herring @phh.

The bug is related to the activation hook. I've not had a proper look into this yet though.

**Comment by phh** [comment]
But I guess [this part](https://github.com/johnbillion/query-monitor/blob/master/query-monitor.php#L36) will make sure that the activation hook is not triggered when calling from cli?

**Comment by johnbillion** [comment]
Correct, but even without it the activation doesn't work. That's a recent
addition.

**Comment by danielbachhuber** [comment]
I haven't looked into it yet, but I believe WP-CLI doesn't call the plugin activation hook.

**Comment by phh** [comment]
I just tested this today. WP CLI is triggering activation hooks. At least this is working for me:

**Comment by johnbillion** [comment]
This was fixed in 2.11.2 but has subsequently caused #206.
